VK_App/VK_App/ViewControllers/FirstBarVC/FriendsTableViewController.swift

```diff
@@ -41,116 +21,128 @@ final class FriendsTableViewController: UITableViewController {
+            filteredFriendsMap = [:]
+            sortedMethod()
+        }
+    }
+
+    private var filteredFriendsMap: [Character: [User]] = [:] {
+        didSet {
+            tableView.reloadData()
+        }
+    }
+
+    private var sortedCharacters: [Character] {
+        filteredFriendsMap.keys.sorted()
+    }
 
     // MARK: Life cycle
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        searchBarDelegate()
-        sortedMethod()
-        infoAboutUser()
+        setMethods()
     }
 
     // MARK: Private Methods
 
-    private func infoAboutUser() {
-        networkAPIService.fetchFriends()
-        networkAPIService.fetchUserPhotos()
-        networkAPIService.fetchGroups()
-        networkAPIService.fetchSearchedGroup(group: Constants.searchGroupName)
+    private func setMethods() {
+        searchBarDelegate()
+        fetchFriends()
+    }
+
+    private func fetchFriends() {
+        networkAPIService.fetchFriends { [weak self] result in
+            switch result {
+            case let .success(users):
+                DispatchQueue.main.async {
+                    self?.users = users
+                    self?.tableView.reloadData()
+                }
+            case let .failure(error):
+                print(error.localizedDescription)
+            }
+        }
     }
 
     private func searchBarDelegate() {
         searchBar.delegate = self
     }
 
     private func sortedMethod() {
-        for users in friends {
-            guard let firstChar = users.friendName.first
-            else { return }
-
-            if sectionsMap[firstChar] != nil {
-                sectionsMap[firstChar]?.append((users.friendName, users.friendImage, users.userPhotoNames))
+        var results = [Character: [User]]()
+        users.forEach {
+            guard let character = $0.firstName.first else { return }
+            if results[character] != nil {
+                results[character]?.append($0)
             } else {
-                sectionsMap[firstChar] = [(users.friendName, users.friendImage, users.userPhotoNames)]
+                results[character] = [$0]
             }
         }
-        sectionTitles = Array(sectionsMap.keys)
-        sectionTitles.sort()
-        filteredFriendsMap = sectionsMap
+        filteredFriendsMap = results
     }
 
     private func setDidSelectRow(indexPath: IndexPath) {
         let storyboard = UIStoryboard(name: Constants.storyName, bundle: nil)
         guard let friendsPhotosViewController = storyboard
             .instantiateViewController(withIdentifier: Constants.storyFriendPhotoID) as? FriendsPhotosViewController
         else { return }
-        guard let list = filteredFriendsMap[sectionTitles[indexPath.section]]?[indexPath.row] else { return }
-        friendsPhotosViewController.configure(nameFriend: list.0, allPhotoFriend: list.2)
+        guard let friends = filteredFriendsMap[sortedCharacters[indexPath.section]]?[indexPath.row] else { return }
```

Comment:
```suggestion
        guard let friends = filteredFriendsMap[sortedCharacters[indexPath.section]]?[indexPath.row] else { return }
```
с 90 по 93 строку сделай один guard из 2х